src/parsing/Lexer.h

```diff
@@ -0,0 +1,50 @@
+#ifndef ILANG_LEXER_H
+#define ILANG_LEXER_H
+
+#include <string>
+#include <map>
+
+using std::string;
+using std::map;
+
+enum class Token {
+    EOF_ = -1,
+    VAR = -2,
+    END = -3,
+    IF = -4,
+    THEN = -5,
+    ELSE = -6,
+    FOR = -7,
+    WHILE = -8,
+    LOOP = -9,
+    IN = -10,
+    REVERSE = -11,
+    ROUTINE = -12,
+    IS = -13,
+    TYPE = -14,
+    RECORD = -15
+};
+
+class Lexer {
+    int current_token;
+    string identifier;
+    double number;
+    map <char, int> op_priority;
+    map <string, Token> keyword_map;
+
+public:
+    Lexer();
+    Lexer(const Lexer &other);
+
+    int current_token() const;
+    string identifier() const;
+    double  number() const;
+
+    void next();
+
+    int token_pr();
+    void add_op_pr(const char &op, const int priority);
+    const map<char, int> op_priority() const;
```

Comment:
std::map is a red-black tree. Since we use char as a key, it better to make it std::unordered_map which is a hashmap